src/odemis/acq/stream/_live.py

```diff
@@ -423,9 +438,36 @@ def _compensateShift(self):
         logging.debug("Update metadata for SEM image shift")
         self._detector.updateMetadata({MD_POS_COR: self._shift})
 
-    def _onActive(self, active):
+    @isasync
+    def prepare(self):
+        """
+        Perform calibration if needed
+        """
+        logging.debug("Preparing stream %s ...", self)
+        # actually indicate that preparation has been triggered, don't wait for
+        # it to be completed
+        self._prepared = True
+        f = self._executor.submit(self._DoPrepare)
+
+        return f
+
+    def __del__(self):
+        self._executor.shutdown(wait=False)
+
+    def _updateStatus(self):
```

Comment:
This is absolutely ugly to put the '...' trick here. Set the status to 'Automatic SEM alignment in progress' (without any period) for the whole time, and add this "..." trick to the GUI code.